### PR TITLE
issue 15: moving sprite relative to the center of the screen

### PR DESCRIPTION
**Pull request by JustMeGaaRa**

### CODE REVIEW

File: src/Rhendaria.Web/Controllers/PlayerController.cs [hunk 1]
```diff
@@ -42,8 +34,8 @@ public async Task<IActionResult> MovePlayer(string username, [FromBody] Movement
             var zone = _client.GetGrain<IZoneActor>("zone");
 
             Vector2D position = await player.Move(command.Direction);
-
-            Task.Run(() => zone.RoutePlayerMovement(player));
+            await zone.RoutePlayerMovement(player);
```
Comment (SGVoid) [inline src/Rhendaria.Web/Controllers/PlayerController.cs]: Isn't better to use `.ContinueWith` for sending out of updates but not await `RoutePlayerMovement` ?

File: src/Rhendaria.Web/wwwroot/js/game.js [hunk 2]
```diff
@@ -1,6 +1,22 @@
-﻿let app = new PIXI.Application({ width: 500, height: 500 });
+﻿(function () {
+    let response = fetch("pavlo.hodysh", { method: "GET" });
```
Comment (SGVoid) [inline src/Rhendaria.Web/wwwroot/js/game.js]: Is it hardcode ?

File: src/Rhendaria.Web/wwwroot/js/game.js [hunk 3]
```diff
@@ -1,6 +1,22 @@
...
+    });
+    let player = Player.create(
+        Sprite.create(
+            response.player.nickname,
+            response.player.positionX,
+            response.player.positionY
+        ));
+    let game = Game.create(zone, viewport, player, sprites);
+
+    console.log(game);
```
Comment (SGVoid) [inline src/Rhendaria.Web/wwwroot/js/game.js]: Do we need this console.log ?

File: src/Rhendaria.Web/wwwroot/scripts/domain.ts [hunk 4]
```diff
@@ -0,0 +1,132 @@
+﻿class Vector {
```
Comment (SGVoid) [inline src/Rhendaria.Web/wwwroot/scripts/domain.ts]: Isn't better to put these classes into separate files?

File: src/Rhendaria.Web/Scripts/rendering.ts [hunk 5]
```diff
@@ -0,0 +1,81 @@
+﻿/// <reference path="../node_modules/@types/pixi.js/index.d.ts" />
+
+/**
+ * game view setup and initialization
+ **/
+async function loadGameView(nickname: string) {
+    const url = `http://localhost:59023/api/player/${nickname}`;
```
Comment (SGVoid) [inline src/Rhendaria.Web/Scripts/rendering.ts]: We need some way to configure the endpoint

File: test/Rhendaria.Engine.Tests/CollissionDetectingServiceTests.cs [hunk 6]
```diff
@@ -9,120 +9,120 @@ namespace Rhendaria.Engine.Tests
 {
     public class CollissionDetectingServiceTests
     {
```
Comment (SGVoid) [inline test/Rhendaria.Engine.Tests/CollissionDetectingServiceTests.cs]: What wrong with these tests?

File: src/Rhendaria.Web/Controllers/PlayerController.cs (hunk 1 above)
Comment (JustMeGaaRa) [reply src/Rhendaria.Web/Controllers/PlayerController.cs]: Actually, await is transpiled into a ContinueWith-like structure. I believe the source code has changed in latest commits, so this is not relevant anymore.

File: src/Rhendaria.Web/wwwroot/js/game.js (hunk 3 above)
Comment (JustMeGaaRa) [reply src/Rhendaria.Web/wwwroot/js/game.js]: Was removed.

File: src/Rhendaria.Web/wwwroot/scripts/domain.ts (hunk 4 above)
Comment (JustMeGaaRa) [reply src/Rhendaria.Web/wwwroot/scripts/domain.ts]: There's no need, I guess. The project is not that big, so organizing it further is a bit overkill.

File: src/Rhendaria.Web/wwwroot/js/game.js (hunk 2 above)
Comment (JustMeGaaRa) [reply src/Rhendaria.Web/wwwroot/js/game.js]: Changed in latest commits.

File: src/Rhendaria.Web/Startup.cs [hunk 7]
```diff
@@ -30,7 +37,9 @@ public void ConfigureServices(IServiceCollection services)
             services.AddMvc().SetCompatibilityVersion(CompatibilityVersion.Version_2_1);
 
             services.AddSwaggerGen(ConfigureSwagger);
-            services.AddSingleton(new ClusterClientsFactory(Configuration).CreateInstance());
+            services.AddSingleton(new ClusterClientsFactory(Configuration).CreateInstance())
+                .AddTransient<PlayerMovementService>()
```
Comment (SGVoid) [inline src/Rhendaria.Web/Startup.cs]: Don't we need an abstraction for `PlayerMovementService` ?

File: src/Rhendaria.Web/Scripts/game.ts [hunk 8]
```diff
@@ -0,0 +1,174 @@
...
+
+class Sprite {
+    private constructor(
+        public readonly nickname: string,
+        public readonly color: number,
+        public readonly actual: Vector,
+        public readonly relative: Vector,
+        public readonly view: PIXI.Graphics = null) {
+
+        if (view) {
```
[the remark this answers is not in the document]
Comment (JustMeGaaRa) [reply src/Rhendaria.Web/Scripts/game.ts]: This is constructor parameter.

File: src/Rhendaria.Web/Scripts/rendering.ts (hunk 5 above)
Comment (JustMeGaaRa) [reply src/Rhendaria.Web/Scripts/rendering.ts]: I guess this is out of scope, but a valid point.

File: test/Rhendaria.Engine.Tests/CollissionDetectingServiceTests.cs (hunk 6 above)
Comment (JustMeGaaRa) [reply test/Rhendaria.Engine.Tests/CollissionDetectingServiceTests.cs]: I f*cked them up :D Will fix them, simply forgot.

File: src/Rhendaria.Abstraction/Vector2D.cs [hunk 9]
```diff
@@ -5,14 +5,38 @@ namespace Rhendaria.Abstraction
...
 
-        public int Left { get; }
+        public double X { get; }
 
-        public int Top { get; }
+        public double Y { get; }
+
+        public static Vector2D Empty = new Vector2D(0, 0);
+
+        public static Vector2D XAxis = new Vector2D(1, 0);
```
[the remark this answers is not in the document]
Comment (JustMeGaaRa) [reply src/Rhendaria.Abstraction/Vector2D.cs]: Will fix this,

File: src/Rhendaria.Web/Startup.cs (hunk 7 above)
Comment (JustMeGaaRa) [reply src/Rhendaria.Web/Startup.cs]: I don't know, do we? :)

**Pull request merged**